[BE] refactor: 리뷰 상세 조회 dto 개선

Kimprodp opened the pull request:
<!-- 제목: [BE/FE/All] (기능 등) -->
<!-- 아래에 이슈 번호를 매겨주세요 -->

- resolves #731 

---

### 🚀 어떤 기능을 구현했나요 ?
- 리뷰 상세 조회 응답 dto에서 불필요한 부분을 제거했습니다.
- 이전에 조회 로직 개선에 대해 연구하면서 발견했던 부분인데, QuestionAnswerResponse 이후 각 답변에 대한 내용이 text와 options 들로만 구성되도 괜찮다 생각합니다.
- dto에 불필요한 뎁스가 많아질수록 개선이 힘들 것 같아요.

### 🔥 어떻게 해결했나요 ?
- optionGroup에 있는 항목들을 전부 사용하지 않아 optionGroupReponse가 있는 자체가 불필요한 뎁스를 늘리는 것 같아 제거했습니다.
- optionItemResponse 에서도 isChecked 부분을 고정으로 treu로 사용하고 있어 제거했습니다.


#### **변경된 응답 예시**
```json
{
  "formId" : 1,
  "revieweeName" : "아루",
  "projectName" : "리뷰미",
  "createdAt" : "2024-08-01",
  "sections" : [ {
    "sectionId" : 1,
    "header" : "프로젝트 기간 동안, 아루의 강점이 드러났던 순간을 선택해주세요.",
    "reviews" : [ {
      "questionId" : 1,
      "required" : true,
      "questionType" : "CHECKBOX",
      "questionContents" : "프로젝트 기간 동안, 아루의 강점이 드러났던 순간을 선택해주세요.",
      "options" : [ {
        "optionId" : 1,
        "content" : "커뮤니케이션, 협업 능력 (ex: 팀원간의 원활한 정보 공유, 명확한 의사소통)"
      }, {
        "optionId" : 2,
        "content" : "문제 해결 능력 (ex: 프로젝트 중 만난 버그/오류를 분석하고 이를 해결하는 능력)"
      }, {
        "optionId" : 3,
        "content" : "시간 관리 능력 (ex: 일정과 마감 기한 준수, 업무의 우선 순위 분배)"
      } ],
      "answer" : null
    }, {
      "questionId" : 2,
      "required" : true,
      "questionType" : "TEXT",
      "questionContents" : "위에서 선택한 사항에 대해 조금 더 자세히 설명해주세요.",
      "options" : null,
      "answer" : "나산초의 답변"
    } ]
  } ]
}
```





### 📝 어떤 부분에 집중해서 리뷰해야 할까요?
- 놓치고 있는 부분이 있는지 확인해주세요.

### 📚 참고 자료, 할 말
- 해당 PR은 차주 프론트 협의 후 병합합니다~

Kimprodp committed:
refactor: response 불필요한 뎁스 및 항목 제거

Kimprodp committed:
refactor: response 불필요한 뎁스 및 항목 제거

Kimprodp committed:
refactor: ReviewDetailMapper response 수정 사항 반영

Kimprodp committed:
refactor: ReviewDetailMapper response 수정 사항 반영

Kimprodp committed:
test: 테스트에 수정 사항 반영

Kimprodp committed:
test: 테스트에 수정 사항 반영

hyeonji1220 reviewed (approved):
api문서가 바뀌는 것이니 프론트에게 잘 전달해야겠네요
덕분에 조회 로직에서 몇 줄 줄어든 것도 좋네요👏👏

donghoony reviewed (approved):
💯 프론트 먼저 머지하고 진행합시다!

nayonsoso commented:
제안 드립니다🙋🏻‍♀️ 

1/ questions -> reviews
지금은 question 이라는 객체에 질문 내용과 응답 내용이 다 들어가 있어요.
질문과 응답을 담는 객체의 이름은 questions 보다 reviews 가 더 적절할 것 같아요

2/ (1번이 반영된다면) content -> questionContents or questionName
지금의 questions 가 reviews 로 바뀐다면 
지금의 content 는 questionContent 또는 questionName로 바꿔서 "질문의 내용" 이라는 것을 드러내야 할 것 같아요.

nayonsoso reviewed (changes requested):
제안 드립니다🙋🏻‍♀️

1/ questions -> reviews
지금은 question 이라는 객체에 질문 내용과 응답 내용이 다 들어가 있어요.
질문과 응답을 담는 객체의 이름은 questions 보다 reviews 가 더 적절할 것 같아요

2/ (1번이 반영된다면) content -> questionContents or questionName
지금의 questions 가 reviews 로 바뀐다면
지금의 content 는 questionContent 또는 questionName로 바꿔서 "질문의 내용" 이라는 것을 드러내야 할 것 같아요.

Kimprodp committed:
refactor: dto 필드명 변경

Kimprodp committed:
refactor: dto 필드명 변경

donghoony commented:
필드 이름 바뀌면 확실하게 전달해야겠네요~

nayonsoso reviewed (changes requested):
안녕하시렵니까 테드?👋 반영된 것 확인했습니다. 💯

그리고 api 문서의 완성도를 조금 더 높이자면, 
API 문서에서 응답 데이터를 
"하나의 섹션에 첫번째 카테고리 질문, 두번째 서술형 질문" 으로 내려주면 더 좋을 것 같아요~
api 문서에서 두가지 경우에 각각 응답이 어떻게 오는지 볼 수 있도록요!

Kimprodp committed:
docs: 리뷰 상세조회 api 문서 수정

Kimprodp committed:
docs: 리뷰 상세조회 api 문서 수정

nayonsoso reviewed (approved):
구구구구굳~ 수고했어요😊

chysis committed:
design: 아코디언 header의 모든 영역을 클릭 가능하도록 수정 (#939)

chysis committed:
[All] README 업데이트 (#927)

* Update README.md

* fix: 이미지 정렬 및 변경, 이모지 추가

* fix: 형광펜 이미지 교체

nayonsoso committed:
[BE] cd-fix: 배포 시 새로운 이미지를 pull하도록 수정 (#947)

* chore: 새로운 이미지 다운 명령어 추가

* chore: 새로운 이미지 다운 명령어 위치 변경

- shell script 가 아니라 github actions 으로 이동

nayonsoso committed:
chore: 도커 태그 지정 방식 변경 (#948)

- 도커 태그를 하드코딩으로 설정했다.
- profile 과 docker tag 가 다른 이슈가 있었다.

hyeonji1220 committed:
[BE] 예외의 잘못된 로그레벨 수정 및 사용하지 않는 예외 제거 (#951)

* fix: 로그 레벨 info로 조정

* refactor: 사용하지 않는 예외 삭제

donghoony committed:
[BE] docs: 리드미에 인프라 구조 추가 (#952)

donghoony committed:
[BE] Question 패키지를 Template 하위로 이동 (#956)

* refactor: Question 도메인을 Template 아래로 이동

- 전체적인 경계를 아래와 같이 나눔:
1. 템플릿 (질문, 옵션, 섹션, 템플릿)
2. 리뷰 (답변)
3. 하이라이트
4. 리뷰 그룹

* refactor: TemplateService에서 리뷰 그룹 Repository 계층을 참조하지 않도록 수정

donghoony committed:
[All] fix: `[Release]` PR 머릿말 디스코드 알림 처리 (#940)

* fix: [Release] 머릿말에도 디스코드 알림 처리

* refactor: `Release`인 경우 `All`로 처리

donghoony committed:
[BE] refactor: Config 패키지 분리, 스타일 적용 (#957)

* chore: use `@Slf4j` instead of `LoggerFactory.getLogger`

* refactor: `WebConfig`가 하나의 역할만 하도록 수정

* style: dto 불필요한 개행 제거

* refactor: Config는 상위 패키지, Resolver/Interceptor는 각 도메인에 두기

donghoony committed:
[All] chore: add CODEOWNERS (#959)

donghoony committed:
[BE] feat: 하이라이트 벌크 삽입 추가 (#961)

* feat: 하이라이트 벌크 삽입 추가

* refactor: 불필요한 `@Repository` 제거

* fix: `findAll` 사용하지 않도록 수정

* fix: 테스트 수정

* refactor: `NamedParameterJdbcTemplate` 사용하도록 수정

donghoony committed:
[BE] refactor: 하이라이트 매핑 로직 간소화 (#962)

* refactor: 하이라이트 매핑 로직 간소화

* refactor: `Fragments` 변환 로직 dto 내부로 이동

* refactor: 내부 값 사용하도록 변경

* refactor: Stream 가독성 향상

* style: 개행 삭제

* test: 하이라이트가 안 쳐지는 라인이 잘 변환되는지 테스트

Co-authored-by: Hyeonji <110809927+skylar1220@users.noreply.github.com>

* refactor: 메서드 네임 확실하게 수정

---------

Co-authored-by: Hyeonji <110809927+skylar1220@users.noreply.github.com>

donghoony committed:
[BE] refactor/fix: 하이라이트 검증 객체 책임 이동, 하이라이트 삭제 로직 수정 (#966)

* refactor: `HighlightValidator` 삭제

* chore: add todos

* refactor: 타입에 따라 검증하는 객체 이름에 `Typed-`를 접두사로 추가

* feat: `AnswerValidator`에서 정답과 관련된 검증 진행

* refactor: `List` -> `Collection` 받도록 수정

* fix: 리뷰 그룹에 해당하는 하이라이트만 삭제하도록 수정

* refactor: 사용하지 않는 상수 삭제

* refactor: `TypedAnswerValidatorFactory`로 수정

* refactor: 쿼리 수정, 테스트 추가

donghoony committed:
[All] fix: Release PR 코드리뷰 시 디스코드 알림 (#969)

BadaHertz52 committed:
[FE] fix : 리뷰 모아보기 쿼리 캐시로 인한 변경된 형광펜 데이터 반영 오류 수정 (#976)

* docs: 질문별 모아보기 목 데이터 주관식 답변 변경

* feat : 현재 sectionId sessionSotrage에 저장하고, 모아보기 페이지 언마운트 시 삭제하는 기능 추가

* feat : 형광펜 API 요청 성공 후, 해당 질문 쿼리 무효화하는 기능 추가

* feat : 형광펜 데이터 로컬 스토리지에 저장한 코드 삭제

* feat : ReviewCollectionPage 에서 clearEditorAnswerMapStorage 코드 삭제

Kimprodp committed:
[BE] refactor: 2차 정렬 기준 추가 (#972)

* refactor: 2차 정렬 기준 추가

* refactor: 2차 정렬 기준 추가

donghoony committed:
[BE] feat: Spring Cache 적용 (#792)

nayonsoso committed:
[BE] refactor: question 타입과 답변 내용 일치 검증 및 매핑 로직 개선 (#958)

* refactor: 타입에 대한 분기를 answerMapper 안에서 처리할 수 있도록

* refactor: 사용하지 않는 코드 제거

* refactor: 하나의 테스트에 한가지를 테스트하도록

* refactor: 추상화를 더욱 활용하도록

* style: 개행 통일

* refactor: getQuestionType 없애고 바로 supports를 사용하도록

* refactor: 다른 유형의 답변이 동시에 입력되는지에 대한 검증 삭제

* refactor: 함수 이름 변경

* style: 띄어쓰기 추가

* refactor: 접근제한자 변경

* refactor: 접근제한자 변경

- 팀의 컨벤션을 위해 & 유지보수 편한 코드를 위해 접근 제한자를 public & private 으로 통일

* refactor: AnswerMapper 변경

- abstract class -> interface

* fix: 컴파일 에러 해결

- abstract class -> interface 변환 하면서 extends -> implements 변경되지 않은 부분 변경

nayonsoso committed:
[BE] refactor: 요청 제한 관련 코드 제거 (#983)

* refactor: request limit 코드 제거

* refactor: 로컬, 테스트의 request limit 변수 제거

donghoony committed:
[BE] fix: redis 의존성 제거 (#985)

donghoony committed:
cd: develop 오라클 인스턴스에서 실행하도록 수정 (#984)

nayonsoso committed:
[BE] refactor: port config 파일 경로 이동

- Related to 1b6ee74

nayonsoso committed:
chore: prod cd에 사용될 runner 변경 (#992)

- 오라클 러너를 사용하도록 변경

nayonsoso committed:
chore: release cd를 돌릴 수 있는 브랜치 한정 (#1004)

donghoony committed:
[BE] refactor: 테스트 환경에서 캐싱을 사용하도록 하여 환경 통일 (#1007)

hyeonji1220 committed:
refactor: review group에서 사용되는 예외 패키지 이동 (#999)

- review 패키지에서 사용되는 예외도 있으나, review가 reviewGroup을 의존하는 것이 반대의 경우보다 자연스럽기때문에 이동

nayonsoso committed:
[BE] fix: 잘못된 캐시 키 정정 (#1003)

* fix: 잘못된 캐시 키 정정

* fix: 잘못된 캐시 키 정정

- reviewGroupId 로

* refactor: cache 삭제

donghoony committed:
[BE] chore: H2를 테스트에서만 사용하도록 설정 (#1010)

* chore: H2 TestRuntimeOnly 설정, Swagger 의존성 삭제

* chore: 외부 의존성 버전 명시

* chore: 외부 의존성 버전 명시 제거

donghoony committed:
[BE] refactor: TemplateMapper와 ReviewGroup 결합 느슨하게 수정 (#1011)

donghoony committed:
[BE] refactor: DTO에서 Enum 사용하도록 수정 (#1012)

* style: 개행 수정

* refactor: Dto에 Enum 직접 사용하도록 수정

* style: 개행 수정

ImxYJL committed:
chore: 타입, 인터페이스의 이름을 PascalCase로 강제하는 eslint 규칙 추가 (#1034)

BadaHertz52 committed:
feat : ErrorSuspense에 fallback 수정 (#1027)

- Suspense fallback을 props로 받을 수 있게 하면서, 기존의 ErrorBoundary fallback props명 수장

BadaHertz52 committed:
[FE] feat : 받은, 작성한 리뷰 없을 때 보여주는 돋보기 컴포넌트 생성 (#1024)

* feat : EmptyContent 컴포넌트 생성

* refactor: width, height 조건문 리팩토링

BadaHertz52 committed:
[FE] feat : 회원/비회원 리뷰 링크 생성 폼  (#1031)

* feat : 회원 여부에 따른 리뷰 링크 생성폼 변경

* fix : 공백 검사 대상에 공백으로 이루어진 경우도 추가

* feat : ErrorSuespenseContainer에 Suespen fallback props 추가

* refactor: constant/index.ts 에 route 추가

* feat: 리뷰 링크 생성 버튼 컴포넌트로 분리 및 로딩,에러 처리 추가

* feat : 토스트 메세지 핸들러 props 옵셔널로 변경

* feat : 리뷰 생성 API 오류 시, 토스트 띄우는 것으로 변경

* refactor: useURLGeneratorState 생성

- 리뷰 링크 input 상태 관리 훅

* feat : ErrorSuspense에 fallback 수정

- Suspense fallback을 props로 받을 수 있게 하면서, 기존의 ErrorBoundary fallback props명 수장

* fix : isPending을 사용해, 리뷰 생성 중 보여주기

* refactor: InputField setValue props 변경

- setValue -> updateValue

* refactor: 유효성 검사 관련 상수 constants/review 로 이동

* refactor: ReviewNameField -> ReviewGroupDataField 이름 변경

* test : usePostDataForReviewRequestCode 목 핸들러에 props 추가

* chore: 사용하지 않는 주석 삭제

* style : 토스트 위치 하단으로 변경

* feat: ReviewGroupDataField 에러 메세지 표시, 초기화 시점 변경

- blur 시 에러 메세지 표시
- change 시 에러 메세지 초기화

* feat : 비밀번호 입력 시 에러 메세지 초기화

* feat : 공백만 입력한 경우에 대한 오류 메세지 추가

* refactor: props 타입명 앞에 대문자 변경

* chore : 링크 생성 실패 시 오류 실험 시 만든 코드 복구

- 리뷰 링크 생성 실패 시 ErrorBoundary가 아닌 토스트로 에러 표시하기 위해 했던 실험 관련 코드들 이전으로 복구

* fix : 토스트 duration 오류 수정

chysis committed:
[FE] feat: 사용자 프로필 정보 및 프로필 탭 구현 (#1035)

* feat: 사용자 프로필 컴포넌트 구현

* feat: 프로필 탭 컴포넌트 구현 및 관련 타입 작성

* chore: 공통 theme의 zIndex 값 수정

* feat: 프로필 탭 상태 및 외부 요소 클릭 시 닫히게 하는 로직 구현

* feat: 로그인 탭에 들어가는 요소를 관리하는 훅 구현

* chore: 커스텀 훅 적용 및 프로필 요소 드래그 불가능하도록 수정

* feat: 프로필 정보 및 프로필 탭 반응형 구현

* chore: 프로필 사진 src가 없는 경우 회색 배경을 보여주도록 수정

* chore: 프로필 탭의 모든 요소가 드래그 할 수 없도록 수정

* refactor: 시맨틱 태그 적용

* refactor: 프로필 탭의 element를 별도의 컴포넌트로 분리

* refactor: 프로필 탭 element의 content 타입 변경 및 elementId 속성 추가

* refactor: 프로필 탭 element 렌더링 함수 분리 및 리팩토링

* design: 프로필 사진 크기 수정

* chore: UndraggableWrapper의 min-width 속성 제거

* feat: 프로필 탭 element가 길어질 경우 말 줄임표 처리

* design: 프로필 탭 최대 width 설정

* chore: topbar에 z-index 설정

* fix: click 이벤트 핸들러가 전달되지 않는 문제 수정

* refactor: 프로필 탭 element 정보를 ProfileTab에서 불러와 사용하도록 수정

soosoo22 committed:
[FE] refactor: ReviewCard에서 강점 키워드를 표시하는 ReviewKeyword 컴포넌트 분리 및 불필요한 예시 제거 (#1039)

* fix: 강점 키워드 데이터 서버 응답 형식과 통일

* feat: StrengthKeywordList 컴포넌트 생성

* refactor: ReviewCard 폴더 위치 이동 및 디자인 수정

* refactor: formattedCategories 함수 유틸로 분리 및 목록, 상세 페이지에 적용

* refactor: 작성일 표시 방식을 ReviewDate 컴포넌트를 사용하도록 변경

* design: 상세 페이지 디자인 변경

* chore: 작성일 형식을 '-'에서 '.'으로 변경

* refactor: 배열을 처리하던 함수를 단일 문자열을 포맷팅하는 formatKeyword 함수로 변경

* refactor: ReviewCard에서 강점 키워드를 표시하는 ReviewKeyword 컴포넌트 분리

* refactor: KeywordSection에서 MultipleChoiceAnswer 컴포넌트명으로 변경

* refactor: formatKeyword 함수에서 반환 타입을 명시하지 않도록 변경

* style: css 속성 정렬

* refactor: QuestionAnswerSection을 질문과 답변(객관식/주관식) 구조로 변경

* refactor: MultilineTextViewer 상세 페이지 관련 컴포넌트로 폴더 위치 변경

* refactor: 예시 포함 여부에 따라 포맷팅된 답변 또는 원본 답변 표시

* refactor: MultilineTextViewer 컴포넌트 위치를 src/components/common/으로 복원

chysis committed:
[FE] feat: 뒤로 가기 버튼 구현 (#1040)

* chore: 상단 이동 버튼이 중복으로 들어간 문제 해결

* feat: 뒤로 가기 버튼 구현

* chore: 불필요한 주석 제거

* refactor: 뒤로 가기 버튼을 페이지에 배치할 때 필요한 공통 스타일링 처리

* chore: 누락된 스타일 코드 추가

* refactor: props 네이밍 수정

* chore: BackButton의 type 속성 지정

ImxYJL committed:
[FE] feat: 소셜 연동 로그인 버튼 및 로그인 요청 모달  (#1022)

* feat: LoginButton 컴포넌트트

* feat: GithubLoginButton 컴포넌트트

* chore: component 파일의 index에 GithubLoginButton, LoginButton 추가가

* refactor: style 전용 인터페이스 분리 및 공통 컴포넌트 import 경로 수정정

* chore: modals의 index에 ContentModal 추가가

* feat: LoginRequestModal 컴포넌트트

* refactor: 클릭 핸들러 네이밍 간소화

* chore: components 하위에 login 폴더 생성 및 관련 컴포넌트 이동

* chore: login 폴더 추가 이동

* chore: 주석 삭제

* chore: self-closing 적용

* chore: LoginRequestTitle 타입에서 -type 접미사 삭제

* refactor: useTheme 사용 및 스타일 인터페이스에 Omit 대신 Pick 적용

* feat: 로그인 요청 모달에 에러 메세지 구역 추가

* refactor: LoginButton 문구 수정 및 그에 따른 컴포넌트 구조 수정

* refactor: 스타일 객체의 이름을 명확하게 수정

* refactor: LoginRequestTitleMap 생성

* fix: LoginRequestTitleMap에 readonly 추가

* refactor: LoginRequestTitle 관리 방식을 객체로 변경

* fix: LOGIN_REQUEST_TITLE value를 한국어 안내로 변경

soosoo22 committed:
[FE] feat: 리뷰 링크 관리 페이지와 라우터 설정 추가 (#1032)

* refactor: 리뷰미 로고 글자 간격 및 크기 조정

* refactor: 푸터에 border 추가

* feat: 리뷰 링크 관리 페이지 레이아웃 구현

* feat: 리뷰 링크 관리 페이지 구현 및 레이아웃 적용

* feat: 라우터 설정

* fix: Amplitude 방문 이벤트에 리뷰 링크 관리 페이지 추가

* refactor: ErrorSuspenseContainer로 감싼 후, ReviewLinkManagementPage에서 ReviewLinkDashboard로 분리

* feat: ReviewLinkLayout 컴포넌트 추가 및 반응형 스타일 적용

* feat: 모바일 환경에서 CardList의 max-height를 none으로 설정 및 반응형 세부 수정

* fix:  ReviewCard 컴포넌트 flex-start로 상단 정렬 및 위치 비워두기

* refactor: 리뷰 링크 관리 페이지 경로명 및 폴더명 변경

donghoony committed:
[BE] docs: GitHub 연동, 자신이 만든 리뷰 그룹 목록 조회 API 문서 작성 (#1014)

Kimprodp committed:
[BE] docs: 작성한 리뷰 목록 조회, 회원용 리뷰 그룹 생성, 리뷰 그룹 정보, 리뷰 등록 API 문서를 작성한다. (#1017)

* feat: 내가 쓴 리뷰 목록 DTO, 컨트롤러 구현

* docs: 내가 쓴 리뷰 목록 api 테스트 문서 작성

* refactor: 기존 리뷰 그룹 생성을 비회원 리뷰 그룹 생성으로 변경

* feat: 회원용 그룹 생성 DTO, 컨트롤러 구현

* docs: 회원용 그룹 생성 api 테스트 문서 작성

* refactor: 리뷰 그룹 정보 조회 api를 회원, 비회원용으로 수정

* refactor: 리뷰 생성 api를 생성자가 회원, 비회원인 경우로 수정

* fix: 생성일자 최신순 정렬 반영

* test: api 테스트에 쿠키 추가

* refactor: 내가 쓴 리뷰 목록 응답에서 member id 제거

* refactor: 회원 리뷰 작성을 비회원과 통합

* refactor: 리뷰 그룹 생성 api를 회원, 비회원 통합

* refactor: 받은/작성한 리뷰 목록 uri 구체화

* refactor: 내가 만든 그룹 uri 수정

* refactor: 내가 작성한 리뷰에 대해 written -> authored로 표현 수정

---------

Co-authored-by: hyeonjilee <skylar12200@gmail.com>

BadaHertz52 committed:
ci : 환경변수 추가, 프리플라이트 추가 (#1052)

- sentry, amplitude 환경변수 추가
- 환경변수 확인하는 프리플라이트 추가

ImxYJL committed:
[FE] 작성한 리뷰를 확인할 수 있는 반응형 레이아웃 (#1038)

* feat: 현재 미디어 쿼리 상태와 디바이스 종류(boolean)를 리턴하는 훅

* feat: 작성한 리뷰 페이지의 분할 레이아웃을 담당하는 WrittenReviewItem 레이아웃 컴포넌트

* feat: 임시 WrittenReviewList 컴포넌트

* feat: 임시 DetailedWrittenReview 컴포넌트

* feat: 임시 작성한 리뷰 확인 페이지

* feat: 작성한 리뷰 페이지에 대한 임시 라우팅

* feat: 임시 레이아웃, 반응형 적용

* feat: 선택한 리뷰가 없을 때의 컴포넌트 추가

* refactor: 페이지 레이아웃 이름을 더 직관적이고 단순하게 수정

* chore: WrittenReviewPage의 layout 폴더 위치를 component 하위로 변경

* refactor: 작성한 리뷰 확인 페이지의 이름을 WrittenReviewPage로 간략하게 변경

* refactor: 반응형 레이아웃을 위해 queryString 도입 (+변경된 페이지명에 따른 추가 변경사항)

* chore: amplitude 페이지 정보에 작성한 리뷰 확인 페이지 추가

* refactor: 작성한 리뷰 확인 페이지에 early return 스타일 적용

* refactor: useSearchParamAndQuery의 매개변수 paramKey를 optional로 변경

* refactor: 미디어 쿼리 관련 훅 리팩토링 - mediaType 대신 breakpoint로 명시

1. 훅 이름 변경
2. 변수명 변경

* chore: 간단한 변수명 수정

* chore: Breakpoints 타입 분리

* chore: 경로 수정

* refactor: resize 함수에 debounce 추가

* refactor: xSmall 사이즈를 430으로 변경(일단 아이폰 프로 맥스를 위해 430으로 맞춤)

* chore: 에러 바운더리에 적용된 속성 이름 변경

* fix: useDeviceBreakpoints훅에서 large 크기일 때 undefined가 나오던 문제

* refactor: ReviewList와 ReviewListItem 스타일 조정 - large 사이즈에서만 2분할되도록 변경

* refactor: 작성한 리뷰 확인 페이지 반응형 조절 - 2분할은 large 사이즈에서만 이루어지도록 변경

* fix: WrittenReviewList에 고정 width값(Detailed의 Outline과 동일) 부여

soosoo22 committed:
[FE] feat: NavigationTab 컴포넌트 구현 (#1037)

* feat: NavigationTab 컴포넌트 구현

* feat: NavigationTab 반응형 구현

* feat: URL 쿼리 파라미터 값에 따라 탭 인덱스 업데이트

* feat: 세션 스토리지를 활용해서 선택한 탭 상태 유지

* feat: 상태를 추가하여 transition 활성화 여부 제어

* style: css 속성 정렬

* fix: sessionStorage에 activeTab이 없을 경우 기본값 0 설정

* refactor: useState와 세션 스토리지 로직 제거 및 URL 기반으로 활성화 탭 관리

* feat: NavigationTab 존재 시 헤더 border-bottom none 처리

* feat: ROUTE 상수에 리뷰 링크 관리, 작성한 리뷰 확인 페이지 경로 추가

* feat: 작성한 리뷰 확인 페이지 amplitude 방문 이벤트 이름 추가

* feat: 리뷰 링크 페이지에 NavigationTab 적용

* refactor: min-height을 ReviewLinkDashboardContainer로 이동 및 flex-start 적용

BadaHertz52 committed:
[FE] feat : 이미지 스켈레톤 적용 (#1055)

* feat : 이미지 스켈레톤 컴포넌트 구현

- 이미지 컴포넌트를 children으로 받아서, 로드 전에는 스켈레톤을 띄우는 컴포넌트 구현

* feat : 리뷰 연결 페이지 이미지에 스켈레톤 적용

* style : 이미지 스켈레톤 스타일 변경

- 스켈레톤 배경색 변경
- 오타 수정

* feat : 홈페이지 캐러셀에 이미지 스켈레톤 적용

- 슬라이드 너비에 따라 이미지 사이즈 구하는 훅(useSlideImgSize) 생성
- 슬라이드 이미지 스타일 컴포넌트 분리

* chore : 불필요한 주석 삭제 및 경로 수정

* refactor :  가독성을 위해 줄 바꿈 추가

* feat : ImageSkeleton에서 img onLoad 확장할 수 있게 변경

* fix : 스켈레톤 event 타입으로 인한 테스트 오류 수정

- 테스트 시, event타입이 any로 명시되어서 오류가 나는 것으로 추측 -> 타입을 명시해 오류 해결

BadaHertz52 committed:
[FE] refactor : 리뷰 링크 관련 컴포넌트,훅,유틸 위치 변경 (#1057)

* refactor : CopyButton 폴더 위치 components/common으로 변경

* refactor : 리뷰 링크 생성 관련 컴포넌트 위치 변경

- components/reviewURL 폴더로 이동
- URLGeneratorForm/style.ts 파일명 변경 (styles.ts -> style.ts)
- URLGeneratorForm/style.ts에서 InputField, PasswordField 스타일 분리

* refactor : validateInput 폴더 위치 utils 하위로 변경

* feat : 리뷰 리퀘스트 코드 목서버 통일

- 리뷰 연결 페이지에서 사용 가능한 리뷰 리퀘스트 코드를 리뷰 링크 생성 시 발급하도록 변경

* fix : mock data 변경으로 인한 테스트 실패 수정

* refactor : reviewURL 하위에 form 폴더 추가 및 파일 이동

* refactor : components/common에서 ReviewCard export

* chore : 병합과정에서 위치 변경된 useSlideImgSize 위치 변경

* chore : URLGeneratorButton 이벤트 핸들러명 수정

* refactor : FormBody,FormLayout 삭제에 따른 URLGeneratorForm 구조 변경

- FormBody,FormLayout의 스타일을 URLGeneratorForm 로 이동
- 시맨틱 태그를 사용 URLGeneratorForm 구조를 보다 명확히 함

* refactor : reviewURL 폴더 정리 (URLGeneratorForm에 사용하는 컴포넌트들 폴더 이동)

- URLGeneratorForm에 사용하는 컴포넌트들을 해당 폴더의 components 폴더로 이동

nayonsoso committed:
[BE] chore: profile에 따라 다른 H2 의존성 옵션 추가 (#1047)

* chore: profile에 따라 다른 H2 의존성 옵션 추가

* refactor: gradle 환경변수로 h2 의존성 분기

* chore: H2 의존성 옵션 implementation로 변경

donghoony committed:
[BE] refactor: Template 연관관계 재설정 (#1042)

* refactor: N:M 관계를 1:N으로 수정, 중간 테이블 삭제

* refactor: Question - OptionGroup, Item 리팩터링

* fix: 리팩터링 후 컴파일 에러 해결

* refactor: TemplateMapper Root entity에서 접근하도록 수정

* refactor: Template으로부터 중간 데이터 뽑아오도록 수정

* refactor: OnSelectedOptionId 매핑

* fix: DB Initializer 수정

* refactor: 불필요한 fixture 삭제

* refactor: 섹션 이름 기능을 Template 아래로 이동

* refactor: Template fixture 삭제, Template으로 접근할 수 있도록 TemplateRepository만 사용 (초안)

* refactor: Template 이외 Repository 삭제 준비

* fix: Template 패키지 컴파일 오류 해결

* fix: Template 사용하도록 수정

* fix: Template 직접 참조하지 않고 테스트 가능하도록 수정

* refactor: 불필요한 테스트 삭제, 컴파일 오류 해결

* refactor: SelectionCount 값 객체로 분리

* refactor: 불필요한 mappedBy 삭제

* fix: MapperTest 컴파일 오류 해결

* refactor: EqualsAndHashoCde 적용, 프로덕션 코드 가독성 향상

* fix: 올바르지 않은 식 수정

* fix: JPQL Join 구문 수정

* fix: 테스트 의존성 제거, 올바르게 돌아가도록 수정

* refactor: 테스트 명확하게 수정

* fix: 테스트 통과하도록 템플릿 미리 저장

* fix: 테스트 통과하도록 수정

* refactor: remove unused imports

* refactor: DTO 매핑 진행 및 Mapper 삭제

* refactor: Section 이름 DTO 매핑

* refactor: SelectionRange로 이름 변경

* refactor: SelectionRange로 이름 변경

* refactor: 커스텀 예외

* style: 개행 적용

* test: 불필요한 픽스처 사용 제거

* style: 개행 수정

* refactor: 객체 파라미터로 받도록 메서드 수정

* refactor: hasIdOf 제거

* chore: 사용하지 않는 메서드/클래스 제거

* style: 해요체로 변경

* refactor: isCheckbox로 변경

* chore: 사용하지 않는 변수 제거

* feat: OneToOne 연관관계 설정, FlyWay 스크립트 작성

* feat: 참조 키를 Question이 가지도록 수정

* refactor: 선택을 0개부터 할 수 있도록 수정

* refactor: 변수 할당 뒤 범위 검증

* chore: TODO 추가

* chore: OnSelectedOptionId 제약조건 추가

donghoony committed:
[BE] fix: Flyway 스크립트 DDL 오류 수정 (#1071)

Kimprodp committed:
refactor: 변경 사항 미반영분 반영

Kimprodp committed:
Merge remote-tracking branch 'origin/be/refacotr/846-review-detail-response-refactor' into be/refacotr/846-review-detail-response-refactor

# Conflicts:
#	backend/src/main/java/reviewme/review/service/dto/response/detail/QuestionAnswerResponse.java
#	backend/src/test/java/reviewme/api/TemplateFixture.java
